lock-api/src/main/java/com/palantir/lock/watch/FailureCheckingLockWatchEventCache.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+import com.google.common.reflect.AbstractInvocationHandler;
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+
+final class FailureCheckingLockWatchEventCache extends AbstractInvocationHandler {
```

Comment:
nit: Maaaybe consider a different name, because we also rely on the synchronized being here for correctness! First thing that came to my mind was "ResilientLockWatchEventCache", or "SynchronizedLockWatchEventCache".

Reply:
Went with the former.